x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportGetLifecycleAction.java

```diff
@@ -98,9 +98,11 @@ protected void masterOperation(Task task, Request request, ClusterState state, A
                             return;
                         }
                         LifecyclePolicyMetadata policyMetadata = entry.getValue();
-                        if (Regex.simpleMatch(name, entry.getKey())) {
+                        String policyName = entry.getKey();
+                        if (Regex.simpleMatch(name, policyName)) {
+                            policyMetadata.getPolicy().maybeAddDeprecationWarningForFreezeAction(policyName);
                             policyResponseItemMap.put(
-                                entry.getKey(),
+                                policyName,
                                 new LifecyclePolicyResponseItem(
```

Comment:
I'm not sure about having this on the "get lifecycle" API. This API is more likely to be hit in an automated way, over and over again (like a system checking for whether a lifecycle is installed), and that could cause a ton of logging. I think we should stick with just warnings for the "put" side. What do you think?

Reply:
I agree, I was trying to capture "old" policies that are not getting updated, but you are right, this is a pain, so it's probably not worth it